pkg/controllers/provisioning/binpacking/packer.go

```diff
@@ -74,7 +75,7 @@ type Packing struct {
 // It follows the First Fit Decreasing bin packing technique, reference-
 // https://en.wikipedia.org/wiki/Bin_packing_problem#First_Fit_Decreasing_(FFD)
 func (p *Packer) Pack(ctx context.Context, schedule *scheduling.Schedule, instances []cloudprovider.InstanceType) []*Packing {
-	defer metrics.Measure(packDuration)()
+	defer metrics.Measure(packDuration.WithLabelValues(ctx.Value("provisioner").(string)))()
```

Comment:
consider making a helper function to avoid casting inline. Maybe something like:
`inject.GetResource(ctx)`
`injection.WithResource(ctx)`
There are some other helpers that you could move into this pattern like
`injection.GetRestConfig(ctx)` and `injection.GetOptions`


Check out the pattern defined by knative in `/cmd/webhook/main.go` `injection.WithNamespaceScope`.

Reply:
Good idea. I created one under utility. There are a few similar helper functions under utility already. I learned from that pattern.